src/UI/LocationListView.re

```diff
@@ -7,18 +7,10 @@ open Oni_Model;
 
 module Option = Utility.Option;
 
-// TODO: move to Revery
-let getFontAdvance = (fontFile, fontSize) => {
-  open Revery.Draw;
-
-  let scaledFontSize =
-    Text._getScaledFontSizeFromWindow(Revery.UI.getActiveWindow(), fontSize);
-  let font = FontCache.load(fontFile, scaledFontSize);
-  let shapedText = FontRenderer.shape(font, "x");
-  let Fontkit.{advance, _} =
-    FontRenderer.getGlyph(font, shapedText[0].glyphId);
-
-  float(advance) /. 64.;
+let getFontAdvance = (fontFamily, fontSize) => {
+  let window = Revery.UI.getActiveWindow();
+  Revery.Draw.Text.measure(~window, ~fontSize, ~fontFamily, "x").width
+  |> float_of_int;
 };
```

Comment:
The reason it is duplicated is to get more precision. The conversions back and forth between int and float in `measure` and its helper functions causes a loss of precision. Although with the text rendering we have now I understand this will be a whole integer anyway, I'm not convinced that will be the case if we do sub-pixel rendering. It also seems odd that advance has a precision of 64 units per pixel internally if there is no use for it.

Reply:
Right - unfortunately the text rendering in Revery isn't perfect. In particular, the kerning is clamped to integer pixel coordinates (so we don't take full advantage of the [font metrics / precision from FreeType2](https://www.freetype.org/freetype2/docs/tutorial/step2.html)).

>  Although with the text rendering we have now I understand this will be a whole integer anyway, I'm not convinced that will be the case if we do sub-pixel rendering for example.

True! We should probably push this change to Revery - have the `measure` and similar APIs return floating-point to prepare for that (even if it is only integer-precision today). Once Skia is in place, we'll leverage the APIs it has for measuring- like this one: https://github.com/manuhornung/reason-skia/blob/d26e3ddfb7e8cee462d4290f3c29d5298baf72b0/src/Skia.rei#L77 - which would give us floating-point precision. The text rendering provided by Skia is significantly better and higher fidelity than what we have today...

> It also seems odd that advance has a precision of 64 units per pixel internally if there is no use for it.

For sure... I'm looking forward to being out of the text-rendering business and letting Skia handle it...

In the meantime - I'll bring back the previous logic, and a note that we should push the `float` to the upstream measure APIs, and just port over the high-DPI fix here too.

Reply:
Tracking the revery API change here: https://github.com/revery-ui/revery/issues/670